ffi.el

```diff
@@ -98,8 +98,8 @@ SLOT-NAME is a symbol and TYPE is an FFI type descriptor."
 (defmacro define-ffi-array (name type length &optional docstring)
   ;; This is a hack until libffi gives us direct support.
   (let ((type-description
-	 (apply #'ffi--define-struct (make-list (eval length)
-						(symbol-name type)))))
+	 (apply #'ffi--define-struct
+                (make-list (eval length) type))))
```

Comment:
It possible should be `symbol-value' instead of `symbol-name', making user defined types are also valid for array definition

I'll check later on